### DISCUSSION TITLE
Include all commonly used plugins

### DISCUSSION THREAD

**Discussion by caribou-code** [semantic-release/semantic-release#3313] [Ideas] [answered]
By default, semantic release ships with 4 plugins:
- @semantic-release/commit-analyzer
- @semantic-release/release-notes-generator
- @semantic-release/npm
- @semantic-release/github

Is there any reason other commonly used ones are not included in this list? I find myself adding the following plugins every time I use semantic release:
- @semantic-release/changelog
- @semantic-release/git

Looking at package popularity, these appear to have high usage. Initially I found it confusing to understand why some plugins had to be installed manually and others didn't, and what that meant in terms of configuring them and how they're supported.

**Comment by travi** [comment]
> Is there any reason other commonly used ones are not included in this list?

the default configuration is our recommended config for publishing to the npm registry from a repo hosted on github, our most common use case. in that scenario, we [actively](https://semantic-release.gitbook.io/semantic-release/support/faq#why-is-the-package.jsons-version-not-updated-in-my-repository) [discourage](https://semantic-release.gitbook.io/semantic-release/support/faq#should-release-notes-be-committed-to-a-changelog.md-in-my-repository-during-a-release) using the git or changelog plugins.

> I find myself adding the following plugins every time I use semantic release:
> 
> * @semantic-release/changelog
> * @semantic-release/git

is there any reason you are choosing to include those plugins every time you use semantic-release, despite the default configuration and our recommendation against their use?



> Initially I found it confusing to understand why some plugins had to be installed manually and others didn't, and what that meant in terms of configuring them and how they're supported.

our docs can always use improvement. if you have suggestions to make these recommendations more clear, or if you found recommendations elsewhere that contradict our recommendations that we could attempt to influence, we welcome such feedback.

**Reply by caribou-code** [reply]
My company use Azure DevOps instead of Github, so the `git` plugin is required for us. I guess it would be the same for anyone using an alternative platform such as Bitbucket or Gitlab. I expect Github is the most popular, but I'm wondering if there's a particular reason not to bundle the `git` plugin as well, even if it's not included in the default config?

As for the `changelog` plugin, this is an extremely common approach. For example 6 of the top 10 most popular npm packages use this approach, including packages like React and Moment. Again, it doesn't need to be part of the default config but could be bundled with the main `semantic-release` package for ease of use?

**Reply by travi** [reply, marked as answer]
anyone can choose to use either plugin despite our recommendations. there absolutely are valid reasons to use them. that is why they exist. however, they are not our recommendation when they are not needed. they are simply dependencies, like any other. if you choose to include them, part of that choice is the complexity of getting them downloaded for use.

because of that, we purposely do not include them as default dependencies. we will not include default dependencies that are not used in the default config for many reasons, one of which is the weight of unnecessary downloads. if anything, we need to take steps to include fewer plugins by default rather than more. as you mention, you don't use github, so it would be better for you if we did not include that plugin by default.
